lightning/src/routing/router.rs

```diff
@@ -181,7 +215,20 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, targ
 		return Err(LightningError{err: "Cannot generate a route of more value than all existing satoshis".to_owned(), action: ErrorAction::IgnoreError});
 	}
 
-	// We do a dest-to-source Dijkstra's sorting by each node's distance from the destination
+	// The general route finding idea is the following:
+	// 1. Fill first/last hops communicated by the caller.
+	// 2. Construct and store a path.
+	//	Remember which channels were used so that future paths don't rely on the same capacity.
+	//	Remember the used channel with minimum capacity ("bottleneck"), so that future paths avoid using it.
```

Comment:
A "bottleneck" designates a channel where we consume all its capacity due to previous operation of remembering an usage ?

Reply:
Bottleneck is literally `the used channel with minimum capacity`. Not sure what's unclear here.
1. If the selected path was sufficient, then the choice of bottleneck doesn't matter, because we already can route
2. If the selected path was insufficient, at least one of the channels was lower than required amount. We remember that channel so that we avoid it in the next path finding.

(2) is somewhat redundant with `used_channels_with_amounts_msat`. Maybe `used_channels_with_amounts_msat` can be used in-place instead of the bottlenecks.
I'll see if that's possible.

Reply:
I think I understand, do you agree with  "During a path construction attempt, a bottleneck is defined as the channel for which maximal capacity (`htlc_maximum_msat`) is under required amount such triggering a failure of the construction. The faultive channel is pruned out next path construction attemps " ?

`minimum_capacity` is confusing as I though it was `htlc_minimum_msat`.